src/components/PostDetails.tsx

```diff
@@ -1,117 +1,134 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../api/comments';
+
+type Props = {
+  post: Post | null;
+  setError: (message: string) => void;
+  error: string;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  setError,
+  error,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isNewComment, setIsNewComment] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadApiComments = async (postId: number) => {
+    try {
+      setIsLoading(true);
+      const data = await getComments(postId);
+
+      if (!data.length) {
+        setError('No comments yet');
+      }
+
+      setComments(data);
+    } catch {
+      setError('Unable to get comments');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (post?.id) {
+      loadApiComments(post.id);
+      setError('');
+    }
+  }, [post]);
+
+  const handleDelete = (commentId: number) => {
+    deleteComment(commentId);
+    setComments((current) => {
+      return current.filter(comment => comment.id !== commentId);
+    });
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${post?.id}: ${post?.title}`}
```

Comment:
Just don't render the part that connected to the post if you don't have the post.

Reply:
done